stackslib/src/chainstate/stacks/db/transactions.rs

```diff
@@ -1480,6 +1480,15 @@ impl StacksChainState {
             }
         }
 
+        if !StacksBlock::validate_transactions_static_epoch(&vec![tx.clone()], epoch) {
```

Comment:
This is a somewhat expensive check, and it's already performed by the time this code is called.  Can you instead update `StacksChainState::process_transaction_precheck()` to (1) take a `StacksEpochId` argument, and (2) verify that the `.auth` field is supported in the epoch?  It should be the very first check -- before the call to `tx.verify()`